test/test_jit.py

```diff
@@ -1299,110 +1312,51 @@ def softmax(a):
-        xs, batch = self.rand_batch(batch_size, (True, 4), (False, input_size))
-        hx, h_batch = self.rand_batch(batch_size, (False, hidden_size))
-        cx, c_batch = self.rand_batch(batch_size, (False, hidden_size))
+        def view(a):
+            return a.view([4, -1, 3])
 
-        # input to hidden weights
-        w_xi = torch.rand(input_size, hidden_size)
-        w_xf = torch.rand(input_size, hidden_size)
-        w_xo = torch.rand(input_size, hidden_size)
-        w_xc = torch.rand(input_size, hidden_size)
-        # hidden to hidden weights
-        w_hi = torch.rand(hidden_size, hidden_size)
-        w_hf = torch.rand(hidden_size, hidden_size)
-        w_ho = torch.rand(hidden_size, hidden_size)
-        w_hc = torch.rand(hidden_size, hidden_size)
-        # bias terms
-        b_i = torch.rand(hidden_size)
-        b_f = torch.rand(hidden_size)
-        b_o = torch.rand(hidden_size)
-        b_c = torch.rand(hidden_size)
-
-        ys = [LSTM(xs[j], hx[j], cx[j], w_xi, w_xf, w_xo, w_xc,
-                   w_hi, w_hf, w_ho, w_hc, b_i, b_f, b_o, b_c) for j in range(batch_size)]
-        ybs = LSTM_batch(batch, h_batch, c_batch, w_xi, w_xf, w_xo, w_xc,
-                         w_hi, w_hf, w_ho, w_hc, b_i, b_f, b_o, b_c)
-        self.assertEqual(ys, ybs.examples())
+        xs, batch = self.rand_batch(4, (True, 5), (False, 3))
+        res_batch = view(batch)
+        res = [xs[j].view([1, -1, 3]) for j in range(4)]
+        self.assertEqual(res, res_batch.examples())
 
-    def test_numToTensor(self):
+    def test_batch_cat(self):
         @torch.jit.batch(batch_size=4)
-        def batch_numToTensor(a):
-            a = a + 1
-            return a
+        def cat2(a, b):
+            return torch.cat([a, b], 2)
 
-        def single_numToTensor(a):
-            a = a + 1
-            return a
-
-        a, batch = self.rand_batch(4, ())
-        res_batch = batch_numToTensor(batch)
-        res = [single_numToTensor(a[j]) for j in range(4)]
+        xs, batch = self.rand_batch(4, (True, 5), (False, 3))
+        xs2, batch2 = xs, batch
+        res_batch = cat2(batch, batch2)
+        res = [torch.cat([xs[j], xs2[j]], 2) for j in range(4)]
         self.assertEqual(res, res_batch.examples())
 
-        @torch.jit.script
-        def batch_numToTensor(a):
-            a = a + 1
-            return a
+    def test_batch_sum(self):
+        @torch.jit.batch(batch_size=4)
+        def batch_sum(a):
+            return a.sum()
 
-        graph = torch.to_batch_graph(batch_numToTensor.graph)
-        self.assertExpected(str(graph))
+        xs, batch = self.rand_batch(4, (True, 5), (False, 3))
+        res_batch = batch_sum(batch)
+        res = [xs[j].sum().unsqueeze(0) for j in range(4)]
+        self.assertEqual(res, res_batch.examples())
 
     def test_if_else(self):
         @torch.jit.batch(batch_size=4)
         def batch_if(a, b):
             if a > b:
-                a += b
+                a = a + b
```

Comment:
Was there a regression here or something?

Reply:
After adding `requires_grad=True` when creating input tensors, there is a runtime error "RuntimeError: a leaf Variable that requires grad has been used in an in-place operation."